### DISCUSSION TITLE
How to set bucket policy as download using python?

### DISCUSSION THREAD

**Discussion by Sangram9693** [minio/minio#11394] [Q&A] [answered]
I am trying to set bucket policy. Minio hosted in my server, which is not using aws S3.
I am trying to get download link of an image present in buck1/myimge.png

Can we set buck1 policy as download ? If yes, then how can we set policy using python?

I tried to set_bucket_policy function. But I am getting error like policy is not suitable. May be I am passing wrong JSON for download policy.

Can anyone guide me how to set policy for bucket as download?

I want my download URL should look like below
http://localhost:9000/bucket_name/img_name.png

**Comment by kannappanr** [comment, marked as answer]
You can use https://docs.min.io/docs/python-client-api-reference#set_bucket_policy

a policy like this will work in your case

```
{
 "Statement": [
  {
   "Action": [
    "s3:GetBucketLocation",
    "s3:ListBucket"
   ],
   "Effect": "Allow",
   "Principal": {
    "AWS": [
     "*"
    ]
   },
   "Resource": [
    "arn:aws:s3:::buck1"
   ]
  },
  {
   "Action": [
    "s3:GetObject"
   ],
   "Effect": "Allow",
   "Principal": {
    "AWS": [
     "*"
    ]
   },
   "Resource": [
    "arn:aws:s3:::buck1/*"
   ]
  }
 ],
 "Version": "2012-10-17"
}

```

Please feel free to reach out to us at https://slack.min.io for such questions

**Comment by Sangram9693** [comment]
But this JSON is for aws S3. But I am not using S3 bucket

**Comment by Sangram9693** [comment]
Could you please explain how can I set for my local minio server @kannappanr

**Comment by harshavardhana** [comment]
> Could you please explain how can I set for my local minio server @kannappanr

You should read the documentation with examples and your endpoint set to localhost:9000, 

For Q/A join our slack community @Sangram9693
